templates/internal/appName/rpc/server.go.tpl

```diff
@@ -13,19 +13,45 @@ package {{ stencil.ApplyTemplate "goPackageSafeName" }} //nolint:revive // Why:
 
 import (
 	"context"
+  {{- $additionalImports := stencil.GetModuleHook "internal/rpc/server.additionalImports" }}
+  {{- if $additionalImports }}
+      {{- range $additionalImports -}}
+  {{ . | indent 2 }}
+      {{- end }}
+  {{- end }}
 )
 
 // Server is the actual server implementation of the API.
 //
 // Note that tracing, logging and metrics are already handled for these
 // methods.
 type Server struct{
+  {{- $additionalHandlerState := stencil.GetModuleHook "internal/rpc/server.additionalHandlerState" }}
+  {{- if $additionalHandlerState }}
+  // handler state added my modules
+    {{- range $additionalHandlerState -}}
+  {{ . | indent 2 }}
+    {{- end -}}
+  // end handler state added by modules
+  {{- end }}
+
 	// Place any handler state for your service here.
 }
 
 // NewServer creates a new server instance.
 func NewServer(ctx context.Context, cfg *Config) (*Server, error) {
+  {{- $additionalProperties := stencil.GetModuleHook "internal/rpc/server.additionalProperties" }}
+  {{- if $additionalProperties }}
+    return &Server{
+      // properties added by modules
+    {{- range $additionalProperties -}}
+      {{ . | indent 6 }}
```

Comment:
[note] looking for feedback here. adding lines here can get a little messy since it isn't clear who is responsible for adding the `,` after each property added. i felt it made the most sense to leave that up to the consumer of this hook, so they would end up with something like:

```tpl
{{ define "myProperties" }}
one,
two,
foo,
bar,
{{ end }}

{{ stencil.AddToModuleHook "github.com/getoutreach/stencil-golang" "internal/rpc/server.additionalProperties" (list (stencil.ApplyTemplate "myProperties")) }}
```

i would be curious to hear from others what the most ergonomic API here would be.